qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,160 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from __future__ import annotations
+
+from qiskit.algorithms.observables_evaluator import eval_observables
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.time_evolvers.evolution_problem import EvolutionProblem
+from qiskit.algorithms.time_evolvers.evolution_result import EvolutionResult
+from qiskit.algorithms.time_evolvers.real_evolver import RealEvolver
+from qiskit.opflow import (
+    CircuitOp,
+    PauliSumOp,
+    StateFn,
```

Comment:
I guess we are still dependent here in someways on opflow - is this unavoidable at this point?